ydb/core/blobstorage/dsproxy/dsproxy_strategy_base.cpp

```diff
@@ -385,40 +386,19 @@ void TStrategyBase::Prepare3dcPartPlacement(const TBlobState &state,
                 }
             }
         }
+        if (placed < preferredReplicasPerRealm) {
+            fullPlacement = false;
+        }
     }
+    return fullPlacement;
 }
 
 ui32 TStrategyBase::MakeSlowSubgroupDiskMask(TBlobState &state, const TBlobStorageGroupInfo &info, TBlackboard &blackboard,
         bool isPut, const TAccelerationParams& accelerationParams) {
-    if (info.GetTotalVDisksNum() == 1) {
-        // when there is only one disk, we consider it not slow
-        return 0;
-    }
-    // Find the slowest disk
+    // Find slow disks
     switch (blackboard.AccelerationMode) {
-        case TBlackboard::AccelerationModeSkipOneSlowest: {
-            TDiskDelayPredictions worstDisks;
-            state.GetWorstPredictedDelaysNs(info, *blackboard.GroupQueues,
-                    (isPut ? HandleClassToQueueId(blackboard.PutHandleClass) :
-                            HandleClassToQueueId(blackboard.GetHandleClass)),
-                    &worstDisks, accelerationParams.PredictedDelayMultiplier);
-
-            // Check if the slowest disk exceptionally slow, or just not very fast
-            ui32 slowDiskSubgroupMask = 0;
-            if (worstDisks[1].PredictedNs > 0 && worstDisks[0].PredictedNs > worstDisks[1].PredictedNs *
-                    accelerationParams.SlowDiskThreshold) {
-                slowDiskSubgroupMask = 1 << worstDisks[0].DiskIdx;
-            }
-
-            // Mark single slow disk
-            for (size_t diskIdx = 0; diskIdx < state.Disks.size(); ++diskIdx) {
-                state.Disks[diskIdx].IsSlow = false;
-            }
-            if (slowDiskSubgroupMask > 0) {
-                state.Disks[worstDisks[0].DiskIdx].IsSlow = true;
-            }
-
-            return slowDiskSubgroupMask;
+        case TBlackboard::AccelerationModeSkipTwoSlowest: {
+            return MakeSlowSubgroupDiskMaskForTwoSlowest(state, info, blackboard, isPut, accelerationParams);
```

Comment:
I suggest you making this code a bit more universal -- mark slowest N disks. Instead of patching the code every time.